Cannot mock `GitInfo` using `@MockitoConfig(convertScopes = true)`

### Describe the bug

I want to mock `GitInfo` (from the info extension) in my Test. But although I use `@MockitoConfig`, I get

`Invalid use of io.quarkus.test.InjectMock - the injected bean does not declare a CDI normal scope but: jakarta.inject.Singleton. Offending field is gitInfo of test class class org.acme.BuildInfoResourceTest`.

Here is my example Test class:
```kotlin
package org.acme

import io.quarkus.info.GitInfo
import io.quarkus.test.InjectMock
import io.quarkus.test.junit.QuarkusTest
import io.quarkus.test.junit.mockito.MockitoConfig
import io.restassured.RestAssured.given
import io.restassured.http.ContentType.JSON
import org.hamcrest.CoreMatchers.`is`
import org.junit.jupiter.api.Test
import org.mockito.Mockito

@QuarkusTest
class BuildInfoResourceTest {

    @InjectMock
    @MockitoConfig(convertScopes = true)
    lateinit var gitInfo: GitInfo

    @Test
    fun testBuildInfoEndpoint() {
        Mockito.`when`(gitInfo.latestCommitId()).thenReturn("f123A")
        given()
            .`when`()
            .get("/build-info")
            .then()
            .statusCode(200)
            .contentType(JSON)
            .body(`is`("{\"gitHash\":\"f123A\"}"))
    }
}
```

### Expected behavior

I expect annotation `@MockitoConfig(convertScopes = true)` to work and give me a mock.

### Actual behavior

Attempting to run the test gives the stacktrace below:

```

Failed to create test instance
org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
	at io.quarkus.test.junit.QuarkusTestExtension.initTestState(QuarkusTestExtension.java:789)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:755)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.IllegalStateException: Invalid use of io.quarkus.test.InjectMock - the injected bean does not declare a CDI normal scope but: jakarta.inject.Singleton. Offending field is gitInfo of test class class org.acme.BuildInfoResourceTest
	at io.quarkus.test.junit.mockito.internal.CreateMockitoMocksCallback.getBeanHandle(CreateMockitoMocksCallback.java:105)
	at io.quarkus.test.junit.mockito.internal.CreateMockitoMocksCallback.injectField(CreateMockitoMocksCallback.java:47)
	at io.quarkus.test.junit.mockito.internal.CreateMockitoMocksCallback.afterConstruct(CreateMockitoMocksCallback.java:37)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.test.junit.AbstractTestWithCallbacksExtension.invokeCallbacks(AbstractTestWithCallbacksExtension.java:158)
	at io.quarkus.test.junit.AbstractTestWithCallbacksExtension.invokeAfterConstructCallbacks(AbstractTestWithCallbacksExtension.java:64)
	at io.quarkus.test.junit.QuarkusTestExtension.initTestState(QuarkusTestExtension.java:786)
	... 4 more
```

### How to Reproduce?

Run the test to reproduce. Use the uploaded minimal example project if it helps you. Contains more or less nothing but gradle project and the Test.
Testproject: [test.zip](https://github.com/quarkusio/quarkus/files/15038590/test.zip)


### Output of `uname -a` or `ver`

Linux hostname 6.5.0-10031-tuxedo

### Output of `java -version`

openjdk version "21.0.2" 2024-01-16 LTS OpenJDK Runtime Environment Temurin-21.0.2+13 (build 21.0.2+13-LTS) OpenJDK 64-Bit Server VM Temurin-21.0.2+13 (build 21.0.2+13-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.9.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.5

### Additional information

To me, this looks a lot like #35080 or #38817, but I thought these issues would be fixed by now.

--- a/extensions/info/deployment/src/main/java/io/quarkus/info/deployment/InfoProcessor.java
+++ b/extensions/info/deployment/src/main/java/io/quarkus/info/deployment/InfoProcessor.java
@@ -13,7 +13,7 @@ import java.util.Map;
 import java.util.TimeZone;
 import java.util.stream.Collectors;
 
-import jakarta.inject.Singleton;
+import jakarta.enterprise.context.ApplicationScoped;
 
 import org.eclipse.jgit.api.Git;
 import org.eclipse.jgit.api.errors.GitAPIException;
@@ -122,7 +122,7 @@ public class InfoProcessor {
             valuesProducer.produce(new InfoBuildTimeValuesBuildItem("git", data));
             beanProducer.produce(SyntheticBeanBuildItem.configure(GitInfo.class)
                     .supplier(recorder.gitInfoSupplier(branch, latestCommitId, latestCommitTime))
-                    .scope(Singleton.class)
+                    .scope(ApplicationScoped.class)
                     .setRuntimeInit()
                     .done());
         } catch (Exception e) {
@@ -229,7 +229,7 @@ public class InfoProcessor {
         valuesProducer.produce(new InfoBuildTimeValuesBuildItem("build", data));
         beanProducer.produce(SyntheticBeanBuildItem.configure(BuildInfo.class)
                 .supplier(recorder.buildInfoSupplier(group, artifact, version, time, quarkusVersion))
-                .scope(Singleton.class)
+                .scope(ApplicationScoped.class)
                 .setRuntimeInit()
                 .done());
     }
@@ -251,7 +251,7 @@ public class InfoProcessor {
         valuesProducer.produce(new InfoBuildTimeContributorBuildItem(recorder.osInfoContributor()));
         beanProducer.produce(SyntheticBeanBuildItem.configure(OsInfo.class)
                 .supplier(recorder.osInfoSupplier())
-                .scope(Singleton.class)
+                .scope(ApplicationScoped.class)
                 .setRuntimeInit()
                 .done());
     }
@@ -264,7 +264,7 @@ public class InfoProcessor {
         valuesProducer.produce(new InfoBuildTimeContributorBuildItem(recorder.javaInfoContributor()));
         beanProducer.produce(SyntheticBeanBuildItem.configure(JavaInfo.class)
                 .supplier(recorder.javaInfoSupplier())
-                .scope(Singleton.class)
+                .scope(ApplicationScoped.class)
                 .setRuntimeInit()
                 .done());
     }

